src/Component.js

```diff
@@ -59,31 +62,45 @@ export const DebounceInput = React.createClass({
 
 
   componentWillUnmount() {
-    if (this.notify.cancel) {
-      this.notify.cancel();
+    if (this.notify.flush) {
+      this.notify.flush();
```

Comment:
I figured it probably wasn't too critical when the component is on it's way out, but I'll add it anyway for consistency.  👍